src/utils/optionsdict.h

```diff
@@ -100,7 +101,20 @@ class OptionId {
   const char short_flag_;
 };
 
+class Button {
+ public:
+  Button() { val = std::make_shared<bool>(false); }
+  Button(bool x) { val = std::make_shared<bool>(x); }
+  operator bool() { return *val; }
```

Comment:
I'd do en explicit function `was_clicked()` or `test()` rather than `operator bool`.

Also, to save some lines, maybe `bool test_and_reset()` function.